tools/gyp/pylib/gyp/MSVSVersion.py

```diff
@@ -226,6 +234,15 @@ def _CreateVersion(name, path, sdk_based=False):
   if path:
     path = os.path.normpath(path)
   versions = {
+      '2017': VisualStudioVersion('2017',
+                                  'Visual Studio 2017',
+                                  solution_version='12.00',
+                                  project_version='14.0',
+                                  flat_sln=False,
+                                  uses_vcxproj=True,
+                                  path=path,
+                                  sdk_based=sdk_based,
+                                  default_toolset='v141'),
       '2015': VisualStudioVersion('2015',
```

Comment:
So this is a de-novo patch, not based on the strategy in [`node-gyp:configure.js` ](https://github.com/nodejs/node-gyp/blob/master/lib/configure.js#L158).
Could be done in [`configure`](https://github.com/nodejs/node/blob/master/configure) which generates `node.gypi`